### ISSUE TITLE
Merge release/1.5.1 back into develop (after #860), with reminders

### ISSUE THREAD

**Issue by climbfuji** [JCSDA/spack-stack#861]
**Is your feature request related to a problem? Please describe.**

Reminder:
- remove unnecessary fms versions
- Hercules: change name of mvapich2 modulefile
- also: clean up unused hercules modules, installations etc

**Describe the solution you'd like**

**Additional context**

**Comment by natalie-perlin** [comment]
@climbfuji - FYI, there is an issue with loading spack-stack (gnu) on Hercules for the UFS WM that uses spack-stack/1.5.0. 
https://github.com/ufs-community/ufs-weather-model/issues/1985
The issue has to do with versions of slurm and mvapich2. Is it something that has been resolved for spack-stack/1.5.1? 
The current solution then could be just modifying presequisites in the existing modulefile for /work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.5.0/envs/unified-env-mvap2/install/modulefiles/gcc/11.3.1/stack-mvapich2/2.3.7.lua 
Changing a prerequisite/loading slurm/23.02.6 (now a default) from the previous version should suffice. The alternative could be to change mvapich2 module loading an earlier slurm version.

**Comment by climbfuji** [comment]
@natalie-perlin I fixed this on Hercules in spack-stack 1.5.0. It's a manual fix, but since we won't be updating 1.5.0 anymore (1.5.1 is ready), that's good enough. I ran the following commands successfully on Hercules:
```
./compile.sh hercules "-DAPP=S2SWA -D32BIT=ON -DCCPP_SUITES=FV3_GFS_v17_coupled_p8,FV3_GFS_v17_coupled_p8_ugwpv1" "" gnu YES YES 2>&1 |tee compile.log
./rt.sh  -e -n cpld_control_p8 gnu -a gsd-hpcs 2>&1 | tee rt_cpld_control_p8_gnu.log
```

**Comment by natalie-perlin** [comment]
@climbfuji -thank you!
